rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1958,6 +1959,32 @@ public RMMonitoring getMonitoring() {
         }
     }
 
+    @Override
+    public StringWrapper getNodeThreadDump(String nodeUrl) {
+        RMNode node = checkNodeAliveOrFail(nodeUrl);
+        String threadDump;
+        try {
+            threadDump = node.getNode().getThreadDump();
+        } catch (ProActiveException e) {
+            logger.error("Could not get node thread dump for node " + nodeUrl, e);
+            throw new ThreadDumpNotAccessibleException(nodeUrl, "failed fetching thread dump", e);
+        }
+        logger.debug("Thread dump for node " + nodeUrl + ": " + threadDump);
+        return new StringWrapper(threadDump);
+    }
+
+    private RMNode checkNodeAliveOrFail(String nodeUrl) {
+        if (!this.allNodes.containsKey(nodeUrl)) {
+            logger.warn("Could not get node thread dump for node " + nodeUrl);
+            throw new ThreadDumpNotAccessibleException(nodeUrl, "the node is not managed by the Resource Manager");
```

Comment:
Nice method, it could be then used for other purpose than threadDump so the Exception could be more generic.

Reply:
I agree, method name is quite generic but exception throw is very specific. I'll change that.